src/dsp_tools/utils/xmlupload/resource_create_client.py

```diff
@@ -377,3 +356,13 @@ def _make_uri_value(value: XMLValue) -> dict[str, Any]:
             "@value": value.value,
         },
     }
+
+
+def _assert_string(value: str | KnoraStandoffXml) -> str:
```

Comment:
I think `_assert_is_string()` would be more clear. But perhaps this could be solved more elegantly with https://docs.python.org/3.11/library/typing.html#typing.TypeGuard?

Reply:
Interesting... I didn't know of TypeGuard, thanks for the hint!  
However, in this case, if I understand correctly, with TypeGuard I would need to handle the error case everywhere separately, whereas here I have this logic in one place, which seems convenient.

I think I'll just rename it to `_assert_is_string()`

Reply:
Fine :-)